caches/infinispan/src/test/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProviderTest.java

```diff
@@ -15,138 +15,129 @@
-                                            cache, pathgen, events, decorator, executor, nfsBasePath );
-        provider.init();
+        final String content = "This is a test";
+        final Location loc = new SimpleLocation( "http://foo.com" );
+        final String fname = "/path/to/my/file.txt";
+
+        CountDownLatch latch = new CountDownLatch( 2 );
+
+        new Thread( new WriteFileThread( content, loc, fname, latch ) ).start();
+        new Thread( new DeleteFileThread( loc, fname, latch ) ).start();
+        latchWait( latch );
+
+        assertThat( provider.exists( new ConcreteResource( loc, fname ) ), equalTo( false ) );
     }
 
     @Test
-    @BMScript( "TryToReadWhileWritingTestCase.btm" )
-    @Ignore( "Needs to be in a separate class to avoid BindException on byteman JVM agent")
-    public void testTryToReadWhileWriting()
-            throws IOException, InterruptedException
+    @BMScript( "ConcurrentWriteAndReadFile.btm" )
+    public void ConcurrentWriteAndReadFile()
+            throws Exception
     {
-        new Thread( new WriteThread() ).start();
-        new Thread( new ReadThread() ).start();
-        try
-        {
-            latch.await();
-        }
-        catch ( Exception e )
-        {
-            System.out.println( "Threads await Exception." );
-        }
+        final String content = "This is a test";
+        final Location loc = new SimpleLocation( "http://foo.com" );
+        final String fname = "/path/to/my/file.txt";
+
+        CountDownLatch latch = new CountDownLatch( 2 );
+
+        new Thread( new ReadFileThread( loc, fname, latch ) ).start();
+        new Thread( new WriteFileThread( content, loc, fname, latch ) ).start();
+        latchWait( latch );
+
         assertThat( result, equalTo( content ) );
     }
 
     @Test
-    @BMScript( "TryToWriteWhileReadingWithFileExistedTestCase.btm" )
-    @Ignore( "Needs to be in a separate class to avoid BindException on byteman JVM agent")
-    public void testTryToWriteWhileReadingWithFileExisted()
-            throws IOException, InterruptedException
+    @BMScript( "WriteCopyAndReadNewFile.btm" )
+    public void writeCopyAndReadNewFile()
+            throws Exception
     {
-        new Thread( new ReadThread() ).start();
-        new Thread( new WriteThread() ).start();
-        try
-        {
-            latch.await();
-        }
-        catch ( Exception e )
-        {
-            System.out.println( "Threads await Exception." );
-        }
-        assertNull( result );
+        final String content = "This is a test";
+        final Location loc = new SimpleLocation( "http://foo.com" );
+        final String fname = "/path/to/my/file.txt";
+        final Location loc2 = new SimpleLocation( "http://bar.com" );
+
+        CountDownLatch latch = new CountDownLatch( 2 );
+
+        new Thread( new CopyFileThread( loc, loc2, fname, latch ) ).start();
+        new Thread( new WriteFileThread( content, loc, fname, latch ) ).start();
```

Comment:
From the test method name, I thought there was a read thread in here too...? It might make sense to have a very small javadoc that gives the initial state and expected final state (or result) of the test.